### PR TITLE
#1108 use recognize_path and values_match? to handle composable matchers in RouteToMatcher

### PR DESCRIPTION
**Pull request by jtxyz**
First attempt at tackling #1108 (no tests yet)

### CODE REVIEW

**Comment by samphippen** [comment]
hi @jtxyz sorry I'm only just getting to this. Could you take a look at the failing tests on CI and fix them up? Then we can go ahead with review. Thanks ✨

**Comment by pirj** [comment]
Can you please rebase and add a test that would fail without this patch @jtxyz ?

File: lib/rspec/rails/matchers/routing_matchers.rb [hunk 1]
```diff
@@ -22,18 +22,23 @@ def matches?(verb_to_path_map)
...
               path, query = *verb_to_path_map.values.first.split('?')
-              @scope.assert_recognizes(
-                @expected,
-                { :method => verb_to_path_map.keys.first, :path => path },
-                Rack::Utils.parse_nested_query(query)
-              )
+              @recognized_path = @scope.routes.recognize_path(
+                  path, {
+                  method: verb_to_path_map.keys.first,
+                  extras: Rack::Utils::parse_nested_query(query)
```
Comment (pirj) [inline lib/rspec/rails/matchers/routing_matchers.rb]: Please keep `.` for a method call.

File: lib/rspec/rails/matchers/routing_matchers.rb [hunk 2]
```diff
@@ -22,18 +22,23 @@ def matches?(verb_to_path_map)
...
+              })
             end
+            success && values_match?(@expected, @recognized_path)
           end
 
           def failure_message
-            rescued_exception.message
+            if rescued_exception
+              rescued_exception.message
```
Comment (pirj) [inline lib/rspec/rails/matchers/routing_matchers.rb]: Is it possible to use safe navigation here?

**Comment by pirj** [comment]
Closing as stale. Feel free to reopen if you have time on your hands to wrap it up.

**Pull request closed without merging**